### ISSUE TITLE
Load profile on login / talent spec change (Event)

### ISSUE THREAD

**Issue by homezonebenny** [Breeni/BtWLoadouts#48]
### Load Favorit Profile on Character Login

When you have 90% of your spells as macros so that you have to use **general macros** AND **character macros** for one specc of a class, a profile addon is absolutely needed!

In the past there was this addon called "ActionBarProfiles".
It saved: Macros, Spells, ALL Keybindings (even Pings), Talents
Into a profile. Kinda the same way **BtWLoadouts** is doing it.

On profile load, it **DELETE all existing Macros** and **CREATE all macros stored in the profile** and put it into the action bar.
The same feature BtWLoadouts offers.

But there's a big **issue** with BtWLoadouts compared to ActionBarProfiles.
If you are playing with your BM Hunter, so that "all macro fields" are used for BM Hunter macros.
And you logout and login to your Tank Druid, you **STILL** have all the macros loaded from the BM Hunter...
You need to **manually** open BtWLoadouts and load a specific profile.
In **ActionBarProfiles** you can set a **STAR** Symbol to a specific profile, so that it loads *this specific profile* when you are **respecc** to this *specc** and when you **login** to the character.
So for example if you are a **heal druid** and you specc to **tank druid** BtWLoadouts needs to load the "Favourite Profile" for this specc (Guardian Druid). When you respecc to **heal druid** it loads the **favourite** set to the **Heal Druid Specc**.
On **Character Login**, BtWLoadouts checks which specc is active and **LOAD the favourite profile** set to this specc!

If you look at the **Image below** you can see the **Star Symbol** for Tank Druid.
When I respecc to **Heal Druid** you **DON'T** see the Star at the "Tank Druid" but you **see the Star** at the **Favourite Heal Druid Profile!**
So basically, you are able to set **ONE** favourite profile for **each specc** of a class.
So that the Addon **knows** which profile needs to be loaded as **default** when **login / change specc!**

Here are some examples of the Action Bars:
![image](https://github.com/Breeni/BtWLoadouts/assets/11474669/ffb44d2e-fc70-4f45-99a8-05267ddc1bdc)

**Comment by homezonebenny** [comment]
**It could look like this:**
![image](https://github.com/Breeni/BtWLoadouts/assets/11474669/9f9de8b9-0ce9-4060-8f51-dade73579ba3)

Now BtWLoadouts **know** to **load** this profile on **guardian druid** when **LOGIN** and when **changing the specc** from something else to **guardian druid**!!!

The same for other Spezialisations like Heal Druid or Cat or Moonkin.
This makes it **perfect** when you change your specc (normal via Talent Window) and it **still** loads the **default BtWLoadouts Profile** and **ESPECIALLY** on character **login**.

The **Login Default Profile Load** is absolutely **needed!**
Since everytime you login you have to do this:

- Look which specc you are in right now
- Open BtWLoadouts **manually**
- Select the profile **manually**, which can be **confusing** if you have like 10 profiles for ONE specc... "Which profile was again my **default**???"

This is annoying and time consuming each relog from character to character.

**Comment by homezonebenny** [comment]
### Lua Errors on low level char while loading profile of high level char

When you create a profile with a lvl 60 (or whatever) character like **Guardian Druid** and then you create a new lvl 1 character on the island and you want to **load** the profile of your higher druid, it gives you the lua errors below.
Instead it **should** just load everything "it can" like **creating all macros** and put them into the **action bar** and **TRY** to put all spells / items (non macros) into the action bar too!
Without throwing an error message, just **ignore** the **things** can't put into action bar / **talents** which can't be set (yet).

So you are able to **load** the profile every X level again, to get new spells into the action bar from your high level character who created (saved) this profile.
So you can play the same class again (leveling) without thinking which spell / macro on which action bar field!



### Lua Errors

`Message: Usage: local nodeInfo = C_Traits.GetNodeInfo(configID, nodeID)
Time: Fri Apr 12 09:46:00 2024
Count: 1
Stack: Usage: local nodeInfo = C_Traits.GetNodeInfo(configID, nodeID)
[string "=[C]"]: ?
[string "=[C]"]: in function `GetNodeInfo'
[string "@Interface/AddOns/BtWLoadouts/DFTalents.lua"]:182: in function `isActive'
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:356: in function <Interface/AddOns/BtWLoadouts/Loadouts.lua:341>
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:371: in function <Interface/AddOns/BtWLoadouts/Loadouts.lua:365>
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:378: in function <Interface/AddOns/BtWLoadouts/Loadouts.lua:377>
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:438: in function `GetActiveProfiles'
[string "@Interface/AddOns/BtWLoadouts/Events.lua"]:454: in function `?'
[string "@Interface/AddOns/BtWLoadouts/Events.lua"]:19: in function <Interface/AddOns/BtWLoadouts/Events.lua:18>

Locals: (*temporary) = <function> defined =[C]:-1




Message: Usage: local nodeInfo = C_Traits.GetNodeInfo(configID, nodeID)
Time: Fri Apr 12 09:46:02 2024
Count: 1
Stack: Usage: local nodeInfo = C_Traits.GetNodeInfo(configID, nodeID)
[string "=[C]"]: ?
[string "=[C]"]: in function `GetNodeInfo'
[string "@Interface/AddOns/BtWLoadouts/DFTalents.lua"]:182: in function `isActive'
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:356: in function <Interface/AddOns/BtWLoadouts/Loadouts.lua:341>
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:371: in function <Interface/AddOns/BtWLoadouts/Loadouts.lua:365>
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:378: in function <Interface/AddOns/BtWLoadouts/Loadouts.lua:377>
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:438: in function `GetActiveProfiles'
[string "@Interface/AddOns/BtWLoadouts/Events.lua"]:567: in function `EQUIPMENT_SETS_CHANGED'
[string "@Interface/AddOns/BtWLoadouts/Events.lua"]:462: in function `?'
[string "@Interface/AddOns/BtWLoadouts/Events.lua"]:19: in function <Interface/AddOns/BtWLoadouts/Events.lua:18>

Locals: (*temporary) = <function> defined =[C]:-1




Message: Usage: local nodeInfo = C_Traits.GetNodeInfo(configID, nodeID)
Time: Fri Apr 12 09:46:44 2024
Count: 5
Stack: Usage: local nodeInfo = C_Traits.GetNodeInfo(configID, nodeID)
[string "=[C]"]: ?
[string "=[C]"]: in function `GetNodeInfo'
[string "@Interface/AddOns/BtWLoadouts/DFTalents.lua"]:182: in function <Interface/AddOns/BtWLoadouts/DFTalents.lua:179>
[string "@Interface/AddOns/BtWLoadouts/DFTalents.lua"]:301: in function `activate'
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:616: in function <Interface/AddOns/BtWLoadouts/Loadouts.lua:452>
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:718: in function <Interface/AddOns/BtWLoadouts/Loadouts.lua:716>

Locals: (*temporary) = <function> defined =[C]:-1




Message: Interface/AddOns/BtWLoadouts/DFTalents.lua:18: Usage: local success = C_Traits.RollbackConfig(configID)
Time: Fri Apr 12 09:46:49 2024
Count: 1
Stack: Interface/AddOns/BtWLoadouts/DFTalents.lua:18: Usage: local success = C_Traits.RollbackConfig(configID)
[string "=[C]"]: ?
[string "=[C]"]: in function `RollbackConfig'
[string "@Interface/AddOns/BtWLoadouts/DFTalents.lua"]:18: in function `callback'
[string "@Interface/AddOns/BtWLoadouts/BtWLoadouts.lua"]:2126: in function `Call'
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:102: in function <Interface/AddOns/BtWLoadouts/Loadouts.lua:88>
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:459: in function <Interface/AddOns/BtWLoadouts/Loadouts.lua:452>
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:718: in function <Interface/AddOns/BtWLoadouts/Loadouts.lua:716>

Locals: (*temporary) = <function> defined =[C]:-1
`

**Comment by homezonebenny** [comment]
### Level Up Lua Error

```Lua
Message: Usage: local nodeInfo = C_Traits.GetNodeInfo(configID, nodeID)
Time: Fri Apr 12 10:07:15 2024
Count: 1
Stack: Usage: local nodeInfo = C_Traits.GetNodeInfo(configID, nodeID)
[string "=[C]"]: ?
[string "=[C]"]: in function `GetNodeInfo'
[string "@Interface/AddOns/BtWLoadouts/DFTalents.lua"]:182: in function `isActive'
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:356: in function <Interface/AddOns/BtWLoadouts/Loadouts.lua:341>
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:371: in function <Interface/AddOns/BtWLoadouts/Loadouts.lua:365>
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:378: in function <Interface/AddOns/BtWLoadouts/Loadouts.lua:377>
[string "@Interface/AddOns/BtWLoadouts/Loadouts.lua"]:438: in function `GetActiveProfiles'
[string "@Interface/AddOns/BtWLoadouts/Events.lua"]:670: in function `?'
[string "@Interface/AddOns/BtWLoadouts/Events.lua"]:19: in function <Interface/AddOns/BtWLoadouts/Events.lua:18>

Locals: (*temporary) = <function> defined =[C]:-1

```
